externals/ssl/src/CVMFS-CMakeLists.txt

```diff
@@ -0,0 +1,5 @@
+include ("${EXTERNALS_LIB_LOCATION}/CMake-Register_External_Lib.txt")
+
+set (SSL_ARCHIVE "${SSL_BUILTIN_LOCATION}/ssl_install/lib/libssl.a;${SSL_BUILTIN_LOCATION}/ssl_install/lib/libcrypto.a")
```

Comment:
I think the dependency handling in our CMake scripts can be simplified and made more standard and I would schedule this work for 2.4.

* I don't think there is a good reason to have some dependencies compiled at configure-time and others at build-time. The `bootstrap.sh` script should do everything - configuring and building all the deps, in a specific order.
* The dependencies could all be installed to a subdirectory in the project (e.g. `externals_install`) that is ignored by Git. There are two main benefits to this:
  1. This prefix could then be passed to CMake at once and the standard `find_package` command can be used for picking up each dependency. No need to define new CMake variables like ****_ARCHIVE. 
  2. The contents of `externals_install` can be archived and reused (on the same platform). This would allow you to make different types of builds for CVMFS, during development, without always recompiling the build depends.

I think this could make things simpler and remove a lot of custom boilerplate.